content/python/concepts/iterators/terms/iter.md

```diff
@@ -0,0 +1,75 @@
+---
+Title: 'iter()'
+Description: 'The iter() method is a python built-in function that returns an iterator object for a given argument.'
+Subjects:
+  - 'Computer Science'
+  - 'Data Science'
+Tags:
+  - 'Iterators'
+  - 'OOP'
+  - 'Loops'
+CatalogContent:
+  - 'learn-python-3'
+  - 'paths/data-science'
+---
+The **`iter()`** method is a python built-in function that returns an iterator object for a given `object`.
+
+## Syntax
+
+```pseudo
+iter(object, sentinel[optional])
+```
+
+## Parameters                                                                                                         
+* **object** : Required. An iterable object such as list, str, and tuple.                                                                | 
+* **sentinel** : Optional. If the object is a callable object the iteration will stop when the returned value is the same as the sentinel.<br/>If the second argument, sentinel, is given, then object must be a callable object. The iterator created in this case will call object with no arguments for each call to its __next__() method; if the value returned is equal to sentinel, StopIteration will be raised, otherwise the value will be returned. | 
+
+## Example 1
+The example below demonstrates how the `iter()` function is used in a Python program:
+
+```py
+cars = ['Ferrari', 'Lamborghini', 'Porsche']
+
+cars_iter = iter(cars)
+
+print(next(cars_iter))
+print(next(cars_iter))
+print(next(cars_iter))
+
+#Output
+#Ferrari
+#Lamborghini
+#Porsche
+```
+
+## Example 2 with optional parameter sentinel
+One useful application of the second form of iter() is to build a block-reader.  
+For example, reading fixed-width blocks from a binary database file until the end of file is reached:
+```py
+from functools import partial
+with open('mydata.db', 'rb') as f:
+    for block in iter(partial(f.read, 64), b''):
+        process_block(block)
+```
+
```

Comment:
```suggestion

This will produce the following output:

```

And then let's add a \```shell code block below that looks like this:

```shell
Hello, World! Let's count:
2
3
4
5
6
7
8
9
StopIteration Exception raised. Value returned was equal to sentinel value
```

Reply:
I don't understand this request.

Reply:
@mcuocolo See the changes I applied [here](a66477c094d61e22195b61b43b777487f28fef52) to see what I mean.